Improve accessibility on Rails forms

oliverguenther opened the pull request:
This PR tries to improve accessibility in Rails forms through a number of ways.
1. Use a separate error color for form borders (and label errors) with WCAG compliant color contrast.
2. Replace the \* required pseudo content with a separate span that contains an asterisk by default, and is replaced with an explicit span for impaired users.
3. Make field-specific error messages accessible to users by using a separate span within the label.

Should fix:
https://community.openproject.org/work_packages/20446
https://community.openproject.org/work_packages/20449
https://community.openproject.org/work_packages/20447

![demo_project_-_openproject](https://cloud.githubusercontent.com/assets/459462/12484345/3e630340-c05a-11e5-8095-36e8f755e43e.png)

Followup of https://github.com/opf/openproject/pull/3934 on release/5.0

oliverguenther committed:
Add error handling to form labels

This re-enables an `-error` state for labels when the associated
field is erroneous.

Also shows an error icon next to the label and changes the font style.

oliverguenther committed:
Create error color with higher contrast on whites

Contrast on #FFF: 7.5:1

oliverguenther committed:
Add separate span with error message for the current field

oliverguenther committed:
Refactor required star into separate spans

Seperate spans for visible and impaired users allows us to replace the
spelled out Asterisk from pseudo content with a proper aria-label.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-20446-error-labels.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-20446-error-labels.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

ulferts commented on lib/tabular_form_builder.rb:
```diff
@@ -198,10 +198,29 @@ def label_for_field(field, options = {}, translation_form = nil)
+
+      content << content_tag('p',
+                             I18n.t(:label_field_is_required),
+                             class: 'hidden-for-sighted')
```
Is this necessary if the input that is labeled has the html5 "required" attribute?

oliverguenther replied:
Ideally, no. But with Jaws 11, the required attribute isn't detected for all fields by the screen reader. On newer Jaws, you'll get a double mention of required, so we'll likely want to remove this some time later.

See this comment: https://community.openproject.org/work_packages/20447/activity#activity-16

oliverguenther committed:
Remove explicit required label for impaired users

Denote inputs with `aria-required="required"` instead of a separate span
within the label, which causes double mentions of the field being
required (once through the html5 attribute when the screenreader is
compatible) as well as in the label span.

oliverguenther commented:
@ulferts I've replaced the separate required span for impaired users with `aria-required`.

ulferts commented on lib/tabular_form_builder.rb:
```diff
@@ -128,6 +126,12 @@ def container_wrap_field(field_html, selector, options = {})
 
+  def merge_required_attributes(required, options=nil)
+    if required
+      options.merge!({ required: true, 'aria-required': 'required' })
```
This should read `'aria-required': true`
